packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart

```diff
@@ -0,0 +1,157 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:html' as html;
+
+import 'package:google_sign_in_platform_interface/google_sign_in_platform_interface.dart';
+import 'package:flutter_web_plugins/flutter_web_plugins.dart';
+import 'package:js/js.dart';
+import 'package:meta/meta.dart';
+
+import 'src/generated/gapiauth2.dart' as auth2;
+// TODO: Remove once this lands https://github.com/dart-lang/language/issues/671
+import 'src/generated/gapiauth2.dart' show GoogleAuthExtensions;
+import 'src/load_gapi.dart' as gapi;
+import 'src/utils.dart' show gapiUserToPluginUserData;
+
+const String _kClientIdMetaSelector = 'meta[name=google-signin-client_id]';
+const String _kClientIdAttributeName = 'content';
+
+@visibleForTesting
+String gapiUrl = 'https://apis.google.com/js/platform.js';
+
+/// Implementation of the google_sign_in plugin for Web
+class GoogleSignInPlugin extends GoogleSignInPlatform {
+  GoogleSignInPlugin() {
+    _autoDetectedClientId = html
+        .querySelector(_kClientIdMetaSelector)
+        ?.getAttribute(_kClientIdAttributeName);
+
+    _isGapiInitialized =
+        gapi.inject(gapiUrl).then((_) => gapi.init()).then((_) {
+      isInitialized = true;
+    });
+  }
+
+  Future<void> _isGapiInitialized;
+
+  @visibleForTesting
+  Future<void> get isInitializing => _isGapiInitialized;
+
+  @visibleForTesting
+  bool isInitialized = false;
```

Comment:
Do you really need to track this as a separate boolean value? It seems to just be tracking whether the `Future` above has completed or not.

Reply:
I only need to track this to have something to assert in the tests. I'll see if there's any way of asserting a future being completed or not.

Reply:
`expect(methodThatReturnsAFuture, completes);`

Reply:
Ohhh! I didn't come across that matcher, thanks!

Reply:
(Added the `completes` matcher to the test)